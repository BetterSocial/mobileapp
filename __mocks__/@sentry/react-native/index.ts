export const addBreadcrumb = jest.fn();
