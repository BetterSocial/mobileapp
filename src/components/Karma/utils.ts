export const renderCircleColor = (karma: number) => {
  switch (true) {
    case karma > 87.5:
      return ['#50BA57', '#C7EECA', '#50BA57'];
    case karma > 75:
      return ['#70BF21', '#B3E57F', '#70BF21'];
    case karma > 62.5:
      return ['#A0C527', '#D1EA82', '#A0C527'];
    case karma > 50:
      return ['#D3D20C', '#F3F370', '#D3D20C'];
    case karma > 37.5:
      return ['#E4B92E', '#FFE591', '#E4B92E'];
    case karma > 25:
      return ['#F4833C', '#F9BE9A', '#F4833C'];
    case karma > 12.5:
    default:
      return ['#DF001F', '#F5B0B9', '#DF001F'];
  }
};
