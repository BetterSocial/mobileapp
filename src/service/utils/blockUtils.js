import Toast from 'react-native-simple-toast';

import {blockAnonymous, blockUser} from '../../service/blocking';

const uiBlockPostAnonymous = async(postId, source, reason, message, callback) => {
    const data = {
        postId,
        source,
        reason,
        message,
        };
    let result = await blockAnonymous(data);
    if (result.code === 201) {
        callback()
        Toast.show(
            'The user was blocked successfully. \nThanks for making BetterSocial better!',
            Toast.LONG,
        );
    } else {
        Toast.show('Your report was filed & will be investigated', Toast.LONG);
    }
}

const uiBlockUser = async(postId, userId, source, reason, message, callback) => {
    const data = {
        userId,
        postId,
        source,
        reason,
        message,
    };
    let result = await blockUser(data);
    if (result.code === 200) {
        callback()
        Toast.show(
            'The user was blocked successfully. \nThanks for making BetterSocial better!',
            Toast.LONG,
        );
    } else {
        Toast.show('Your report was filed & will be investigated', Toast.LONG);
    }
}

const uiBlockDomain = async(domainId, reason, message, source, callback) => {
    const dataBlock = {
        domainId,
        reason,
        message,
        source,
    };
    const result = await blockDomain(dataBlock);
    if (result.code === 200) {
        callback();
        Toast.show(
            'The domain was blocked successfully. \nThanks for making BetterSocial better!',
            Toast.LONG,
        );
    } else {
        Toast.show('Your report was filed & will be investigated', Toast.LONG);
    }
    console.log('result block user ', result);
}


export default {
    uiBlockPostAnonymous,
    uiBlockUser,
    uiBlockDomain
}