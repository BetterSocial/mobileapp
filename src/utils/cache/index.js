import AsyncStorage from '@react-native-async-storage/async-storage';

const CACHE_NAME = 'cache';
export const saveToCache = (cacheKey, jsonData) => {
  if (jsonData && cacheKey && typeof jsonData === 'object' && typeof cacheKey === 'string') {
    AsyncStorage.getItem(CACHE_NAME, (err, result) => {
      if (result && typeof result === 'string') {
        let parseResult = JSON.parse(result);
        parseResult = { ...parseResult, [cacheKey]: jsonData };
        AsyncStorage.setItem(CACHE_NAME, JSON.stringify(parseResult));
      } else {
        const dataSave = {
          [cacheKey]: jsonData,
        };
        AsyncStorage.setItem(CACHE_NAME, JSON.stringify(dataSave));
      }
    });
  }
};

export const getSpecificCache = (cacheKey, callback) => {
  AsyncStorage.getItem(CACHE_NAME, (error, result) => {
    if (result && typeof result === 'string') {
      const newResult = JSON.parse(result);
      if (newResult[cacheKey]) {
        callback(newResult[cacheKey]);
      } else {
        callback(null);
      }
    } else {
      callback(null);
    }
  });
};

export const removeAllCache = () => {
  AsyncStorage.clear();
};
