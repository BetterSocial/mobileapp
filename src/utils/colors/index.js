// https://chir.ag/projects/name-that-color/
export const colors = {
  bondi_blue: '#00ADB5',
  white: '#ffffff',
  black: '#0C0C0C',
  emperor: '#4F4F4F',
  gray: '#69707C',
  gray1: '#9B9FA9',
  lightgrey: '#FFFFFF',
  concrete: '#F2F2F2',
  mine_shaft: '#333333',
  bunting: '#11243D',
  alto: '#E8EBED',
  porcelain: '#ecf0f1',
  silver: '#BDBDBD',
  french_pass: '#b6e4fd',
  pattens_blue: '#ddf2fe',
  blue: '#2F80ED',
  lightblue: '#55C2FF',
  blue1: '#4782D7',
  red: '#FF2E63',
  holytosca: '#003848',
  holytosca30percent: '#0038484D',
  blackgrey: '#69707C',
  redalert: '#FF2E63',
  halfBaked: '#88CDD0',
  tradewind: '#58B1B5',
  elm: '#22878B',
  blockColor: '#FF2E63',
  darkBlue: '#4782D7',
  blueSea10: '#55C2FF',
  greenDark: '#004346',
  babyBlue: '#7B9DCD',

  signed_primary: '#4782D7',
  signed_secondary: '#6295DD',
  anon_primary: '#003848',
  anon_secondary: '#154B5B',
  radical_red: '#FF3466',
  light_silver: '#D2D4DB'
};

export const hexToRgb = (hex, opacity = 1) => {
  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  const r = parseInt(result[1], 16);
  const g = parseInt(result[2], 16);
  const b = parseInt(result[3], 16);
  return result ? `rgba(${r},${g},${b},${opacity})` : null;
};

export const checkIsHasColor = (color) => {
  const regex = /^#[0-9a-f]{3,6}$/i;
  return regex.test(color);
};
