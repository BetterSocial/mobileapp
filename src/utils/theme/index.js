export const COLORS = {
  almostBlack: '#16202A',
  alto: '#E0E0E0',
  black: '#FFFFFF', // changed to white, before is #1E1F20
  black000: '#000',
  black30percent: 'rgba(0,0,0,0.3)',
  black80percent: 'rgba(0,0,0,0.8)',
  black43: '#434343',
  blue: '#2F80ED',
  blueLoyal: '#023B60',
  blueOnboarding: '#4782D7',
  blueTanzanite: '#11516F',
  blueSea: '#55C2FF',
  blueZaffre: '#0E24B3',
  brilliance: '#FCFCFC',
  bunting: '#11243D',
  concrete: '#FF00FF', // changed to gray 100, before is #F2F2F2
  emperor: '#4F4F4F', // not used again
  french_pass: '#b6e4fd',
  greenMantis: '#79B45D',
  greyseries: '#333333',
  mine_shaft: '#333333',
  pattens_blue: '#ddf2fe',
  platinum: '#E6E6E6',
  porcelain: '#ecf0f1',
  silver: '#BDBDBD',
  waterspout: '#9DEDF1',
  white: '#16202A', // almost black
  white2: '#FFFFFF',
  white10percent: 'rgba(255,255,255,0.1)',
  white20percent: 'rgba(255,255,255,0.2)',
  whiteSmoke: '#F4F4F4',
  bluePrimary: '#4782D7',

  // need to remove
  holytosca: '#00ADB5',
  lightBlue: '#2F80ED',
  gray1: '#C4C4C4',
  gray100: '#FF00FF',
  gray500: '#FF00FF',
  gray6: '#F2F2F2',
  gray4: '#4A4A4A',
  bondi_blue: '#00ADB5',

  // new color
  // signed_primary: '#4782D7',
  // signed_primary_15: '#4782D726',
  // signed_secondary: '#6295DD',
  signed_primary: '#0391FB',
  signed_primary_15: '#0391FB26',
  signed_secondary: '#275D8A',
  default_signed_secondary: '#0391FB',
  // anon_primary: '#107793',
  // anon_primary_15: '#10779326',
  // anon_secondary: '#3A9BB6',
  anon_primary: '#00ABB2',
  anon_primary_15: '#00ABB226',
  anon_secondary: '#184A57',
  radical_red: '#FF3466',
  light_silver: '#D2D4DB',
  balance_gray: '#9B9FA9',
  gray: '#E8EBED',
  lightgrey: '#FF00FF', // changed to gray 200 #F5F6F7
  redalert: '#FF2E63',
  blackgrey: '#FF00FF', // changed to gray 400
  greySubtile1: '#9B9FA9', // changed to gray 400
  elm: '#22878B',
  halfBaked: '#88CDD0',
  greenDark: '#004346',
  holytosca30percent: '#00ADB54D',
  black50: 'rgba(0, 0, 0, 0.5)',
  black75: 'rgba(0, 0, 0, 0.75)',
  transparent: 'transparent',
  anonSecondary20: 'rgba(0, 173, 181, 0.2)',
  gray400: '#FF00FF',
  lightgrey60: 'rgba(245, 246, 247, 0.6)',
  gray300: '#FF00FF',
  gray200: '#FF00FF',
  downvote: '#E2887E',
  upvote: '#69DD10'
};
export const SIZES = {
  // global sizes
  base: 8,
  font: 14,
  radius: 12,
  padding: 24,

  // font sizes
  h1: 36,
  h2: 16,
  h3: 14,
  h4: 12,
  body1: 24,
  body2: 16,
  body3: 14,
  body4: 12

  // app dimensions
};
export const FONTS = {
  h1: {fontFamily: 'Inter-Black', fontSize: SIZES.h1, lineHeight: 36},
  h2: {fontFamily: 'Inter-Bold', fontSize: SIZES.h2, lineHeight: 30},
  h3: {fontFamily: 'Inter-Bold', fontSize: SIZES.h3, lineHeight: 22},
  h4: {fontFamily: 'Inter-Bold', fontSize: SIZES.h4, lineHeight: 22},
  body1: {fontFamily: 'Inter-Regular', fontSize: SIZES.body1, lineHeight: 36},
  body2: {fontFamily: 'Inter-Regular', fontSize: SIZES.body2, lineHeight: 30},
  body3: {fontFamily: 'Inter-Regular', fontSize: SIZES.body3, lineHeight: 22},
  body4: {fontFamily: 'Inter-Regular', fontSize: SIZES.body4, lineHeight: 22},
  comment: {fontFamily: 'Inter-Regular', fontSize: SIZES.body3, lineHeight: 19}
};

export const hexToRgb = (hex, opacity = 1) => {
  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  const r = parseInt(result[1], 16);
  const g = parseInt(result[2], 16);
  const b = parseInt(result[3], 16);
  return result ? `rgba(${r},${g},${b},${opacity})` : null;
};

export const checkIsHasColor = (color) => {
  const regex = /^#[0-9a-f]{3,6}$/i;
  return regex.test(color);
};

const appTheme = {COLORS, SIZES, FONTS};

export default appTheme;
