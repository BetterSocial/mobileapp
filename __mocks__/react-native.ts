import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN