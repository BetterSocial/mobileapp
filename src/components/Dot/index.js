import * as React from 'react';
import {View, StyleSheet} from 'react-native';
import {COLORS} from '../../utils/theme';

const Dot = ({size = 4, color}) => (
  <View testID="dot" style={[styles.point, {width: size, height: size, backgroundColor: color}]} />
);

export default Dot;

const styles = StyleSheet.create({
  point: {
    borderRadius: 4,
    backgroundColor: COLORS.gray400
  }
});
