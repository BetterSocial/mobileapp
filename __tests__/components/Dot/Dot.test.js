import * as React from 'react';
import { render, cleanup } from '@testing-library/react-native';
import Dot from '../../../src/components/Dot';
import { COLORS } from '../../../src/utils/theme';


describe('Dot should run correctly', () => {
// value, index, select, onSelect, icon, desc
    afterEach(cleanup)
    it('should match snapshot', () => {
  
        const {toJSON, getByTestId} = render(<Dot size={5} color='blue' />)
        expect(toJSON).toMatchSnapshot()
        expect(getByTestId('dot').props.style).toEqual([{ borderRadius: 4,
    backgroundColor: COLORS.gray,}, {width: 5, height: 5, backgroundColor: 'blue'}])
    })
})