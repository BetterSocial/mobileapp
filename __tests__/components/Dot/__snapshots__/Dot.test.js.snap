// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dot should run correctly should match snapshot 1`] = `[Function]`;
